src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -17507,3 +17507,118 @@ def tz_convert(self, *args: Any, **kwargs: Any) -> None:
 
     def tz_localize(self, *args: Any, **kwargs: Any) -> None:
         ErrorMessage.method_not_implemented_error("tz_convert", "BasePandasDataset")
+
+    def create_qc_with_index_data_and_qc_index(
+        self, index_qc: "SnowflakeQueryCompiler"
+    ) -> "SnowflakeQueryCompiler":
+        """
+        This is a helper function for creating a DataFrame/Series where the data is an Index
+        and an index is provided.
+        Before this method is called, the provided index is converted to an Index object;
+        the query compilers of the data and index are then joined.
+
+        Parameters
+        ----------
+        index_qc : SnowflakeQueryCompiler
+            The query compiler of the index to be joined with the data.
+
+        Returns
+        -------
+        SnowflakeQueryCompiler
+            A new query compiler with the data and index joined.
+        """
+        self_frame = self._modin_frame.ensure_row_position_column()
+        other_frame = index_qc._modin_frame.ensure_row_position_column()
+
+        new_internal_frame, _ = join_utils.join(
+            self_frame,
+            other_frame,
+            how="left",
+            left_on=[self_frame.row_position_snowflake_quoted_identifier],
+            right_on=[other_frame.row_position_snowflake_quoted_identifier],
+            inherit_join_index=InheritJoinIndex.FROM_RIGHT,
+        )
+
+        return SnowflakeQueryCompiler(new_internal_frame)
+
+    def create_qc_with_data_and_index_joined_on_index(
+        self,
+        index_qc: Optional["SnowflakeQueryCompiler"] = None,
+        extra_columns: Optional[List[Hashable]] = None,
```

Comment:
Why use an optional list instead of leaving the list empty if there are no extra columns?

Reply:
In parts of the code where the `extra_columns` param doesn't apply, I just leave it out instead of giving it an empty list. I don't think there's any difference functionally?